### PR TITLE
Guid windows 20190307

### PR DESCRIPTION
**Pull request by cfcs**
This is kind of a mess history-wise, but it should incorporate:

- jbuilder -> dune transition ( @yomimono )
- recent 0.7.0 changes
  - ip / ethernet changes ( @hannesm )
- GUI:
  - window support
  - docstrings
- types for the *"RPC filecopy"* API ([basic example here](https://github.com/cfcs/eye-of-mirage/blob/master/command.ml#L15-L61) )

I read through the diff and I think it makes sense, but please let me know if I missed something.
ping @talex5 @yomimono @reynir @hannesm

### CODE REVIEW

File: lib/formats.ml [hunk 1]
```diff
@@ -117,13 +117,8 @@ module GUI = struct
...
   ]
 
-  [%%cstruct
-      type msg_clipboard_data = {
-        len_big : uint32_t [@big_endian];
-        len_little : uint32_t;
-        (* followed by a uint8 array of size len *)
-      } [@@little_endian]
-  ]
+  (** Dom0 -> VM, VM -> Dom0: MSG_CLIPBOARD_DATA:*)
```
Comment (reynir) [inline lib/formats.ml]: What happened to msg_clipboard_data?

Comment (reynir) [reply lib/formats.ml]: Digging in the history I see this was never used anyway.

Comment (cfcs) [reply lib/formats.ml]: Hmm good point, I used it, but it is implemented funny in Qubes upstream anyway so better to wait for the fix I think: https://github.com/QubesOS/qubes-gui-daemon/pull/15

Comment (cfcs) [reply lib/formats.ml]: Hmm good point, I used it, but it is implemented funny in Qubes upstream anyway so better to wait for the fix I think: https://github.com/QubesOS/qubes-gui-daemon/pull/15

**Comment by cfcs** [comment]
Superseded by #37

**Pull request closed without merging**